### DISCUSSION TITLE
Script to sort images into folders based on compression type while retaining folder structure?

### DISCUSSION THREAD

**Discussion by D3SOL4TE** [ImageMagick/ImageMagick#7178] [Help]
### ImageMagick version

7.1.1-29

### Operating system, version and so on

Windows 10

### Description

I need a script to recursively search through a folder and sort images into folders based on their type of compression plus if they have alpha or not while also retaining their folder structure.

I already have the following Windows command to recursively search through a directory and output each images compression/alpha.
```for /r "D:\.Temp\" %a in (*.*) do identify -format "%d %f \| %C \| %[channels] \n" "%~a"```

**Comment by emmayashri** [comment]
Hey hi, try it
```
# Define the root folder
$rootFolder = "D:\.Temp\"

# Recursively search through the root folder
Get-ChildItem -Path $rootFolder -Recurse -File | ForEach-Object {
    # Get image compression type and alpha channels
    $compressionType = (magick identify -format "%C" $_.FullName).Trim()
    $hasAlpha = (magick identify -format "%[channels]" $_.FullName).Trim()

    # Define destination folder based on compression type and alpha channels
    $destinationFolder = Join-Path $rootFolder "$compressionType\$hasAlpha"

    # Create destination folder if it doesn't exist
    if (-not (Test-Path $destinationFolder)) {
        New-Item -ItemType Directory -Path $destinationFolder -Force
    }

    # Move image to destination folder
    Move-Item -Path $_.FullName -Destination $destinationFolder
}

```
before deployment try it and for manually compression try any application such as [Jpeg compressor](https://jpegcompressor.com/)
